### ISSUE TITLE
[Bug]: Unable to get the symbol of a `SimpleNameReferenceNode`

### ISSUE THREAD

**Issue by aneeshafedo** [ballerina-platform/ballerina-lang#37762]
### Description

When trying to get the symbol of a `SimpleNameReferenceNode` of a `ExpressionNode` using `semanticModel.symbol(simpleReferenceNode)` returns `null` as the result. 

**Sample Code**

```bal
string data = check weatherClientLocal->/weather/[country]/weather.get;
```

Trying to get the symbol of the `ComputedResourceAccessSegmentNode` : `[country]` 

### Related area

Other

**Comment by NipunaRanasinghe** [comment]
@dulajdilshan @mohanvive since this is a blocking issue for the SA tool work (which is targeted for the kubecon in next month),  will it be possible to prioritise this fix for the next update 2 patch release (2201.2.1)?

**Comment by dulajdilshan** [comment]
> @dulajdilshan @mohanvive since this is a blocking issue for the SA tool work (which is targeted for the kubecon in next month),  will it be possible to prioritise this fix for the next update 2 patch release (2201.2.1)?

Sure. I have already started working on this

**Comment by aneeshafedo** [comment]
Please find the simplified valid code [here](https://gist.github.com/aneeshafedo/20be42c956edcc3209920c576fc5c1b6). 

```bal
import ballerina/http;

service /sample on new http:Listener(5050) {
    resource function get greeting(string country) returns error|Weather{
        http:Client weatherClientLocal = check new("http://localhost:8080");
        Weather data = check weatherClientLocal->/weather/[country].get;
        return data;
    }
}


public type Weather record {
    # Weather condition id
    int id?;
    # Group of weather parameters (Rain, Snow, Extreme etc.)
    string main?;
    # Weather condition within the group
    string description?;
    # Weather icon id
    string icon?;
};
```

Unable to access the symbol of both `weatherClientLocal` and `country` in the line 
`Weather data = check weatherClientLocal->/weather/[country].get;`
